### PR TITLE
Display version information in dashboard and host

### LINKED ISSUE #2222: Add Aspire version number and/or git SHA to console logging
A user reported an issue with the dashboard and included the console log: https://github.com/dotnet/aspire/issues/1556#issuecomment-1943069117. However, there is no way to tell what exactly what version and what commit the app was built from.

We could write the Aspire version number and git commit SHA to the console when the Aspire host and dashboard start up.

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #2222

```
...
info: Aspire.Hosting.DistributedApplication[0]
      Version: 8.0.0-dev
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application started. Press CTRL-C to stop.
```

and
 
```
2024-02-14T19:00:26.3890044 info: Aspire.Dashboard[0]
      Version: 8.0.0-dev
2024-02-14T19:00:26.3995683 info: Aspire.Dashboard[0]
      Now listening on: https://localhost:15887/
```

Locally it shows limited information but when published the metadata should be complete. I checked with the nuget packages that it's provided. I'll check again when it merged in the nightly feed.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2242)

### CODE REVIEW

File: src/Aspire.Dashboard/DashboardWebApplication.cs [hunk 1]
```diff
@@ -80,9 +81,14 @@ public DashboardWebApplication()
         builder.Services.AddLocalization();
 
         _app = builder.Build();
-
         var logger = _app.Logger;
 
+        if (GetType().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion is string informationalVersion)
+        {
+            // Display version and commit like 8.0.0-preview.2.23619.3+17dd83f67c6822954ec9a918ef2d048a78ad4697
+            logger.LogInformation("Version: {InformationalVersion}", informationalVersion);
```
Comment (JamesNK) [inline src/Aspire.Dashboard/DashboardWebApplication.cs:89]:
```suggestion
            // Write version information at information level so it's written to the console by default to help debug user issues.
            // Display version and commit like 8.0.0-preview.2.23619.3+17dd83f67c6822954ec9a918ef2d048a78ad4697
            logger.LogInformation("Aspire version: {Version}", informationalVersion);
```

File: src/Aspire.Hosting/DistributedApplicationLifecycle.cs [hunk 2]
```diff
@@ -30,6 +31,12 @@ public Task StartedAsync(CancellationToken cancellationToken)
 
     public Task StartingAsync(CancellationToken cancellationToken)
     {
+        if (GetType().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion is string informationalVersion)
+        {
+            // Display version and commit like 8.0.0-preview.2.23619.3+17dd83f67c6822954ec9a918ef2d048a78ad4697
+            logger.LogInformation("Version: {InformationalVersion}", informationalVersion);
```
Comment (JamesNK) [inline src/Aspire.Hosting/DistributedApplicationLifecycle.cs:37]:
```suggestion
            // Write version information at information level so it's written to the console by default to help debug user issues.
            // Display version and commit like 8.0.0-preview.2.23619.3+17dd83f67c6822954ec9a918ef2d048a78ad4697
            logger.LogInformation("Aspire version: {Version}", informationalVersion);
```

**Review by JamesNK — APPROVED** [review]
Looks good. Couple of tweaks:

* Write `Aspire version:`
* Comment about why

**Comment by radical** [comment]
Just wondering - would it be useful to surface these versions in the dashboard UI too?

**Comment by JamesNK** [comment]
It already is in the settings popup.

**Pull request merged**